server/svix-server/src/queue/mod.rs

```diff
@@ -91,33 +91,37 @@ impl MessageTaskBatch {
 #[serde(rename_all = "camelCase")]
 #[serde(tag = "type")]
 pub enum QueueTask {
+    HealthCheck,
     MessageV1(MessageTask),
     MessageBatch(MessageTaskBatch),
 }
 
 impl QueueTask {
-    pub fn msg_id(self) -> MessageId {
+    pub fn msg_id(self) -> Option<MessageId> {
         match self {
-            QueueTask::MessageV1(task) => task.msg_id,
-            QueueTask::MessageBatch(batch) => batch.msg_id,
+            QueueTask::HealthCheck => None,
+            QueueTask::MessageV1(task) => Some(task.msg_id),
+            QueueTask::MessageBatch(batch) => Some(batch.msg_id),
         }
     }
-    pub fn to_msg_task(self, endpoint_id: EndpointId) -> MessageTask {
+    pub fn to_msg_task(self, endpoint_id: EndpointId) -> Option<MessageTask> {
```

Comment:
Yuck...

Also, given that these functions are only used once each (based on your code), I think having the `match` be inline where they are used is MUCH more clear. I'd just have the match at the top of the worker and set *both* msg_id and trigger_type into local variables directly there.